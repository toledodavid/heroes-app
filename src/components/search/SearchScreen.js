import React from 'react';
import { heroes } from '../../data/heros';
import { useForm } from '../../hooks/useForm';
import HeroCard from '../heros/HeroCard';


const SearchScreen = () => {

  const herosFilter = heroes;

  const [formValues, handleInputChange] = useForm({searchText: ''});

  const {searchText} = formValues;

  const handleSearch = (event) => {
    event.preventDefault();
    console.log(searchText);
  }

  return(
    <div>
      <h1>Search Hero</h1>
      <hr/>

      <div className="row">
        <div className="col-5">
          <h4>Search Form</h4>
          <hr/>

          <form onSubmit={handleSearch}>
            <input type="text" name="searchText" value={searchText} onChange={handleInputChange} placeholder="Find your hero" className="form-control" autoComplete="off" />

            <button type="submit" className="btn btn-outline-primary btn-block">Search</button>
          </form>
        </div>

        <div className="col-7">
          <h4>Results</h4>
          <hr/>

          {
            herosFilter.map(hero => (
              <HeroCard key={hero.id} {...hero} />
            ))
          }

        </div>
      </div>

    </div>
  );
}

export default SearchScreen;