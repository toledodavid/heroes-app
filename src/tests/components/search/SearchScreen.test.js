import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { MemoryRouter, Route } from 'react-router-dom';
import SearchScreen from '../../../components/search/SearchScreen';

describe('Tests for <SearchScreen /> component', () => {

  test('It should render correctly with default values', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/search']}>
        <Route path="/search" component={SearchScreen} />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.alert-info').text().trim()).toBe('Search a hero');
  });

  test('It should show Batman and input field shoul have queryString value', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <Route path="/search" component={SearchScreen} />
      </MemoryRouter>
    );

    expect(wrapper.find('input').prop('value')).toBe('batman');
    expect(wrapper).toMatchSnapshot();
  });

  test('It should show an error if hero is not found', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/search?q=batmancwefw']}>
        <Route path="/search" component={SearchScreen} />
      </MemoryRouter>
    );

    expect(wrapper.find('.alert-danger').text().trim()).toBe('There is no hero with batmancwefw');
    expect(wrapper).toMatchSnapshot();
  });

  test('It should call push history', () => {
    const history = {
      push: jest.fn()
    }

    const wrapper = mount(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <Route path="/search" component={() => <SearchScreen history={history} />} />
      </MemoryRouter>
    );

    wrapper.find('input').simulate('change', {
      target: {
        name: 'searchText',
        value: 'batman'
      }
    });

    wrapper.find('form').prop('onSubmit')({
      preventDefault(){}
    });

    expect(history.push).toHaveBeenCalledWith('?q=batman');
  });

});