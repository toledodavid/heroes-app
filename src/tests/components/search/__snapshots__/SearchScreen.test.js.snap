// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <SearchScreen /> component It should render correctly with default values 1`] = `
<div>
  <h1>
    Search Hero
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          className="btn btn-outline-primary btn-block"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;

exports[`Tests for <SearchScreen /> component It should show Batman and input field shoul have queryString value 1`] = `
<div>
  <h1>
    Search Hero
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="batman"
        />
        <button
          className="btn btn-outline-primary btn-block"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Batman"
              className="card-img"
              src="./assets/heroes/dc-batman.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Batman
              </h5>
              <p
                className="card-text"
              >
                Bruce Wayne
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Detective Comics #27
                </small>
              </p>
              <a
                href="/hero/dc-batman"
                onClick={[Function]}
              >
                More...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tests for <SearchScreen /> component It should show an error if hero is not found 1`] = `
<div>
  <h1>
    Search Hero
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="batmancwefw"
        />
        <button
          className="btn btn-outline-primary btn-block"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="alert alert-danger"
      >
        There is no hero with 
        batmancwefw
      </div>
    </div>
  </div>
</div>
`;
